### PR TITLE
[Big refactor] move unets to  `unets` module 🦋

### PR DESCRIPTION
**Pull request by sayakpaul**
# What does this PR do?

As discussed internally, we would like to move the UNet related stuff to their dedicated module `unets` just like we did with `autoencoders`. 

**Note to reviewers**: It's a big refactor. So, please take your time in reviewing it. But sooner we're able to merge this, lesser the number of merge conflicts we'll run into.

### CODE REVIEW

**Comment by sayakpaul** [comment]
Out of the two failing tests, the first should pass as soon as https://huggingface.co/hf-internal-testing/tiny-sdxl-custom-all/discussions/2 is merged. The second one fails even on `main`.

File: src/diffusers/models/unets/unet_2d_blocks.py [hunk 1]
```diff
@@ -0,0 +1,3591 @@
+# Copyright 2023 The HuggingFace Team. All rights reserved.
```
Comment (patrickvonplaten) [inline src/diffusers/models/unets/unet_2d_blocks.py:1]: Why does this file only have 3591 lines but the original deleted `unet_2d_blocks.py` file has 3635 lines?

File: src/diffusers/models/unets/unet_2d.py [hunk 2]
```diff
@@ -17,10 +17,10 @@
 import torch
```
Comment (patrickvonplaten) [inline src/diffusers/models/unets/unet_2d.py:17]: Let's make sure we also keep a `src/diffusers/models/unet_2d.py` file for backward compatibility

**Review by patrickvonplaten — COMMENTED** [review]
Ok for me! Let's make sure that for import files such as `unet_2d_condition.py`, `unet_2d_blocks.py`, we keep the original file and import all classes of the moved files for backwards compatibility

File: src/diffusers/models/unets/unet_2d.py (hunk 2 above)
Comment (sayakpaul) [reply src/diffusers/models/unets/unet_2d.py:17]: Done in a7adbeaa

**Comment by sayakpaul** [comment]
Thanks, @patrickvonplaten. Added a couple classes in `models` for BC. Would you prefer more?

**Review by DN6 — APPROVED** [review]
Fine for me to merge. Failing pipeline test looks unrelated, think just rerunning the workflow should get it to pass.

**Comment by sayakpaul** [comment]
I am going to merge this to avoid merge conflicts.

**Pull request merged**